nixos/modules/misc/nixpkgs.nix

```diff
@@ -244,6 +304,21 @@ in
           message = "The NixOS nixpkgs.pkgs option was set to a Nixpkgs invocation that compiles to target system ${pkgsSystem} but NixOS was configured for system ${nixosExpectedSystem} via NixOS option ${nixosOption}. The NixOS system settings must match the Nixpkgs target system.";
         }
       )
+      {
+        # soft deprecate crossSystem when hostPlatform is set
+        assertion = opt.hostPlatform.isDefined -> !opt.crossSystem.isDefined;
+        message = "If you set hostPlatform, you must only define hostPlatform and/or buildPlatform, but none of the legacy options such as crossSystem.";
```

Comment:
```
let explanation = "If you set hostPlatform, you must only define hostPlatform and/or buildPlatform, but none of the legacy options (localSystem crossSystem or system)";
```
```suggestion
        message = ''do not set crossSystem with hostPlatform. ${explanation}";
```

Reply:
And shouldn't we do for buildPlatform too?

Reply:
Done. It's a single assertion message now. See `test.nix`.